File: vnoj-solutions/thcshuynhvannghe_baker.py
import sys

sys.stdin = open("BAKER.INP", "r")
sys.stdout = open("BAKER.OUT", "w")

n = int(sys.stdin.readline())
a = list(map(int, sys.stdin.readline().split()))
b = list(map(int, sys.stdin.readline().split()))

best = 0
for i in range(1, n):
    if a[i] * b[best] > b[i] * a[best]:
        best = i

print(best + 1)

File: vnoj-solutions/thcshuynhvannghe_baker.cpp
#include<bits/stdc++.h>
#define int long long
using namespace std;

signed main(){
    ios_base::sync_with_stdio(0);cin.tie(0);
    freopen("BAKER.INP","r",stdin);
    freopen("BAKER.OUT","w",stdout);
    int n;
    cin >> n;

    vector<int> a(n), b(n);
    for(int& ele: a) cin >> ele;
    for(int& ele: b) cin >> ele;

    int best = 0;
    for(int i = 1; i < n; ++i){
        if((__int128)a[i] * b[best] > (__int128)b[i] * a[best]){
            best = i;
        }
    }

    cout << best + 1 << '\n';

    return 0;
}
